pkg/render/logstorage/components/elasticsearch_metrics.go

```diff
@@ -0,0 +1,148 @@
+package component
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/tigera/operator/pkg/render/common/secret"
+
+	"github.com/tigera/operator/pkg/ptr"
+	relasticsearch "github.com/tigera/operator/pkg/render/common/elasticsearch"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	operatorv1 "github.com/tigera/operator/api/v1"
+	"github.com/tigera/operator/pkg/components"
+	"github.com/tigera/operator/pkg/render"
+	rmeta "github.com/tigera/operator/pkg/render/common/meta"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+func ElasticsearchMetrics(
+	installation *operatorv1.InstallationSpec,
+	clusterConfig *relasticsearch.ClusterConfig,
+	esMetricsCredsSecret *corev1.Secret,
+	esCertSecret *corev1.Secret,
+	clusterDomain string,
+) render.Component {
+	return &elasticsearchMetrics{
+		installation:         installation,
+		clusterConfig:        clusterConfig,
+		esMetricsCredsSecret: esMetricsCredsSecret,
+		esCertSecret:         esCertSecret,
+		clusterDomain:        clusterDomain,
+	}
+}
+
+type elasticsearchMetrics struct {
+	installation         *operatorv1.InstallationSpec
+	esMetricsImage       string
+	esMetricsCredsSecret *corev1.Secret
+	esCertSecret         *corev1.Secret
+	clusterConfig        *relasticsearch.ClusterConfig
+	clusterDomain        string
+}
+
+func (e *elasticsearchMetrics) ResolveImages(is *operatorv1.ImageSet) error {
+	var err error
+	var errMsgs []string
+
+	reg := e.installation.Registry
+	path := e.installation.ImagePath
+
+	e.esMetricsImage, err = components.GetReference(components.ComponentElasticsearchMetrics, reg, path, is)
+	if err != nil {
+		errMsgs = append(errMsgs, err.Error())
+	}
+
+	if len(errMsgs) != 0 {
+		return fmt.Errorf(strings.Join(errMsgs, ","))
+	}
+	return nil
+}
+
+func (e *elasticsearchMetrics) Objects() (objsToCreate, objsToDelete []client.Object) {
+	toCreate := []client.Object{e.metricsService(), e.metricsDeployment()}
```

Comment:
I think we should have these objects in the order they should be created.
I think that should be:
1. Secret
2. Deployment
3. Service

Reply:
Makes sense, I'll do that.